crates/transaction-pool/src/validate/eth.rs

```diff
@@ -289,6 +291,27 @@ where
             )
         }
 
+        let cap = self.tx_fee_cap();
+        if cap != 0 {
```

Comment:
we also want to check that this transaction is `local` because this setting should only be applied to locals?